### ISSUE TITLE
TryApplyChanges not getting updated actual solution

### ISSUE THREAD

**Issue by kanjareeyalalji** [daveaglick/Buildalyzer#101]
Hi,
currently I am using .net core with Buildalyzer and creating work-space using same and modifying into  project but changes is not reflated in my actual solution.

please find below code:
```
 AnalyzerManager manager = new AnalyzerManager("G:\test\test.sln");
using (var workspace = manager.GetWorkspace())
 {
       var solution = workspace.CurrentSolution;
       var newSolution = solution.AddDocument(DocumentInfo.Create(DocumentId.CreateNewId(solution.ProjectIds.FirstOrDefault()), "Test.cs"));
        workspace.TryApplyChanges(newSolution);
 }
```
it adding document in my workspace, however not in my actual solution.

**Comment by kurtgronbech** [comment]
Same problem for me although I came across it while trying to apply a change to a document

```
var updatedDocument = document.WithText(sourceText);
var success = workspace.TryApplyChanges(updatedDocument.Project.Solution);

```
success returns true, but the document is NOT updated on disk. 

Help?

**Comment by daveaglick** [comment]
Given that both of you have had issues in this area, I probably need to improve the documentation to clarify. Buildalyzer isn’t intended to provide two-way mutability back to the underlying project or solution.

The way it works is to execute MSBuild to gather enough information about the project to create a Roslyn workspace for analysis. That workspace though isn’t “connected” to the project other than that MSBuild told Buildalyzer what source files and libraries should be included. Being able to update the project from the Buildalyzer Workspace would require a lot of extra information about the underlying project and _why_ certain files are included, not to mention understanding how to update an MSBuild project file. Unfortunately this is definitely out of scope for this project and that’s unlikely to change.

TL;DR: Buildalyzer is for analysis only and you can’t update project or solution files from the Roslyn Workspace returned by Buildalyzer.

**Comment by kurtgronbech** [comment]
Thanks, understood. In case others come across this issue later I fixed it by using the Roslyn workspace from Buildalyzer only temporarily in ordre to populate the metadata references in my other Roslyn workspace loaded with MSBuildWorkspace like this: 
```

// Load temp solution with Buildalyzer

manager = new AnalyzerManager(solutionFilePath, options);
AdhocWorkspace tmp_workspace = manager.GetWorkspace();
tmp_solution = tmp_workspace.CurrentSolution;

// Load solution with MSBuildWorkspace
workspace = MSBuildWorkspace.Create(props); 
solution = await workspace.OpenSolutionAsync(solutionFilePath);
			
for (int i = 0; i < solution.Projects.Count(); i++)
{
	Project project = solution.Projects.ToList()[i];

	Project projectFromBuildalyzer = tmp_solution.Projects.FirstOrDefault(a => a.Id == project.Id);
				
	project = project.WithMetadataReferences(projectFromBuildalyzer.MetadataReferences); 							
}
```

It seems to work fine and solves my need for now...

**Comment by daveaglick** [comment]
@kurtgronbech That’s a really cool hybrid idea! I’ll add it as a suggestion to the docs for folks that need to mutate their projects.
